### ISSUE TITLE
Improve lore2sphinx buildbot results for pb core howtos

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4863]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4863|
|Type|enhancement|
|Created|2011-02-11 05:07:03Z|
|Branch|https://github.com/twisted/twisted/tree/pb-sphinx-4863|

like [#4566](https://github.com/twisted/twisted/issues/4566), except for the seven pb howtos in `projects/core/howto`

<details><summary>Searchable metadata</summary>

```
trac-id__4863 4863
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Lore_to_Sphinx Lore to Sphinx
branch__branches_pb_sphinx_4863 branches/pb-sphinx-4863
branch_author__screwtape__jdb screwtape, jdb
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1297400823000000 1297400823000000
changetime__1300090557000000 1300090557000000
version__None None
owner__jonathanj jonathanj
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [30658]) Branching to 'pb-sphinx-4863'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30659]) Merge the pb parts of the impossibly huge [[#4568](https://github.com/twisted/twisted/issues/4568)](https://github.com/twisted/twisted/issues/4568) branch

refs #4568
refs [#4863](https://github.com/twisted/twisted/issues/4863)

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_4"></a>|@jonathanj commented|
|-|-|

(In [30697]) Fix footnote linking, refs [#4863](https://github.com/twisted/twisted/issues/4863).

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_5"></a>|@jonathanj commented|
|-|-|

(In [30698]) Fix unexpected new paragraphs, refs [#4863](https://github.com/twisted/twisted/issues/4863).

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_6"></a>|@jonathanj commented|
|-|-|

I fixed a few minor issues:
 * Broken footnote linking due to unusual markup style in the footnote.
 * Unexpected paragraph breaks around `<code>` items, particularly in footnotes.

This branch looks good to merge.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_8"></a>|@jonathanj commented|
|-|-|

I'm putting this back up for review because I'd like someone with experience with previous lore2sphinx tickets to give it a once over.

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_9"></a>|@jonathanj commented|
|-|-|

Sphinx output [here](http://buildbot.twistedmatrix.com/builds/sphinx-html/sphinx-html-15753/).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

Replying to [jonathanj](#note_8):
> I'm putting this back up for review because I'd like someone with experience with previous lore2sphinx tickets to give it a once over.

What would be the best place to review the lore2sphinx tickets, can't seem to find the wiki page.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_13"></a>|@jesstess set owner to @jonathanj|
|-|-|

Thanks for working on the chunks of this enormous changeset. Some feedback:

There are some issues that exist everywhere in the docs. Maybe they need their own meta-tickets:
 * <code> tags are forcing a space after the enclosed text, even if you don't want it.
 * Similarly, the <q> tags are forcing a space afterwards, even if you don't want it.
 * We've lost the hyperlinks on the listings.

Ticket-specific feedback:
 * pb-cred: the twisted.spread.flavors.Viewable hyperlink is broken (it is on the live documentation too).
 * pb-usage:
  * Needs a space after the 'the': The “client” can give a reference to the”server”
  * Needs a space after the 'and': fuzzes the distinction between “client” and”server"
  * There's a broken hyperlink to twisted.spread.pb.InsecureJelly

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_14"></a>|@jonathanj commented|
|-|-|

(In [31005]) Fix broken API hyperlinks and some whitespace issues, refs [#4863](https://github.com/twisted/twisted/issues/4863).

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_15"></a>|@jonathanj set owner to @jesstess|
|-|-|

Replying to [jesstess](#note_13):
> There are some issues that exist everywhere in the docs. Maybe they need their own meta-tickets:
>  * <code> tags are forcing a space after the enclosed text, even if you don't want it.
>  * Similarly, the <q> tags are forcing a space afterwards, even if you don't want it.
>  * We've lost the hyperlinks on the listings.

I've mentioned some of these in passing before; since they're really issues with the lore2sphinx tool (as far as I can tell) I think they belong in another ticket.

> Ticket-specific feedback:
>  * pb-cred: the twisted.spread.flavors.Viewable hyperlink is broken (it is on the live documentation too).
>  * pb-usage:
>   * Needs a space after the 'the': The “client” can give a reference to the”server”
>   * Needs a space after the 'and': fuzzes the distinction between “client” and”server"
>   * There's a broken hyperlink to twisted.spread.pb.InsecureJelly

Fixed.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)|@khorn set owner to @khorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_17"></a>|@khorn set owner to @jonathanj|
|-|-|

Replying to [jonathanj](#note_15):
> Replying to [jesstess](#note_13):
> > There are some issues that exist everywhere in the docs. Maybe they need their own meta-tickets:
> >  * <code> tags are forcing a space after the enclosed text, even if you don't want it.
> >  * Similarly, the <q> tags are forcing a space afterwards, even if you don't want it.
> >  * We've lost the hyperlinks on the listings.
> 
> I've mentioned some of these in passing before; since they're really issues with the lore2sphinx tool (as far as I can tell) I think they belong in another ticket.

Yes, these are lore2sphinx issues.  It is impractical to fix them automatically.  These will be handled post-conversion.

> 
> > Ticket-specific feedback:
> >  * pb-cred: the twisted.spread.flavors.Viewable hyperlink is broken (it is on the live documentation too).
> >  * pb-usage:
> >   * Needs a space after the 'the': The “client” can give a reference to the”server”
> >   * Needs a space after the 'and': fuzzes the distinction between “client” and”server"
> >   * There's a broken hyperlink to twisted.spread.pb.InsecureJelly
> 
> Fixed.


Verified these have been fixed.  Looks good to me.

Merge like the wind!

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_18"></a>|@jonathanj set status to `closed`|
|-|-|

(In [31014]) Merge branches/pb-sphinx-4863.

Author: jdb, screwtape, jonathanj
Reviewer: khorn, jesstess
Fixes: [#4863](https://github.com/twisted/twisted/issues/4863)

Clean up lore2sphinx conversion for PB core howto documentation.
